demo/NostrSDKDemo/Views/KeyInputSectionView.swift

```diff
@@ -0,0 +1,135 @@
+//
+//  KeyInputSectionView.swift
+//  NostrSDKDemo
+//
+//  Created by Joel Klabo on 8/13/23.
+//
+
+import SwiftUI
+import NostrSDK
+
+enum KeyType {
+    case `public`
+    case `private`
+}
+
+struct KeyInputSectionView: View {
+    
+    @Binding var key: String
+    @Binding var isValid: Bool
+
+    var type: KeyType
+
+    var body: some View {
+        HStack {
+            TextField(label(for: type),
+                      text: $key)
+                .font(.system(size: 16, weight: .regular, design: .monospaced))
+                .autocapitalization(.none)
+                .autocorrectionDisabled()
+
+            if key.isEmpty {
+                EmptyView()
+            } else if isValid {
+                Image(systemName: "checkmark.seal.fill")
+                    .foregroundColor(.green)
+            } else {
+                Image(systemName: "x.circle.fill")
+                    .foregroundColor(.red)
+            }
+        }
+        .onChange(of: key) { _ in
+            isValid = checkValidity(key: key, type: type)
+        }
+        .onAppear {
+            isValid = checkValidity(key: key, type: type)
+        }
+    }
+
+    private func label(for type: KeyType) -> String {
```

Comment:
Add this to the `KeyType` enum itself like this:
```swift
var label: String {
    switch self {
    case .public: return "Public Key"
    case .private: return "Private Key"
    }
}
```

Then the call sites are much cleaner:
```swift
keyType.label
```

Reply:
🤙